client/components/data/query-post-stats/index.jsx

```diff
@@ -16,7 +16,7 @@ function QueryPostStats( { siteId, postId, fields } ) {
 	const memoizedFields = useMemoCompare( fields, ( a, b ) => a?.join() === b?.join() );
 
 	useEffect( () => {
-		if ( siteId && postId ) {
+		if ( siteId && typeof postId === 'number' ) {
```

Comment:
Non-integers will fail on the backend anyway. This one allows querying for `post_id = 0` which is a special case for stats. We can also do `postId > -1`. Both are fine by me.